bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,109 @@
+import asyncio
+import datetime
+import logging
+from typing import List, Optional
+
+from discord import Color, Embed, Message
+from discord.ext import commands
+
+from bot.constants import Wikipedia
+
+log = logging.getLogger(__name__)
+
+SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch={search_term}&format=json"
+WIKIPEDIA_URL = "https://en.wikipedia.org/wiki/{title}"
+
+
+class WikipediaCog(commands.Cog):
+    """Get info from wikipedia."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    @staticmethod
+    def formatted_wiki_url(index: int, titles: str) -> str:
+        """Making formatted wikipedia link.."""
+        return f'`{index}` [{titles}]({WIKIPEDIA_URL.format(title=titles.replace(" ", "_"))})'
+
+    async def search_wikipedia(self, search_term: str) -> Optional[List[str]]:
+        """Search wikipedia and return the first page found."""
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+
+        # Ignore pages with "may refer to"
+        for search_result in search_results:
+            log.info("trying to append titles")
+            if "may refer to" not in search_result["snippet"]:
+                page.append(search_result["title"])
+        log.info("Finished appending titles")
+        return page
```

Comment:
```suggestion
    async def search_wikipedia(self, search_term: str) -> List[str]:
        """Search wikipedia and return the first 10 pages found."""
        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
            data = await response.json()

        pages = []

        search_results = data["query"]["search"]

        # Ignore pages with "may refer to"
        for search_result in search_results:
            log.info("trying to append titles")
            if "may refer to" not in search_result["snippet"]:
                pages.append(search_result["title"])

        log.info("Finished appending titles")
        return pages
```
`page` is a list, so it should have a plural name.
Dont know why you added back Optional after Pure saying it's not right. This function will always return a list, the only difference is that it might be an empty list. Which i dont think you need to worry about typing.
The docstring says "first page", but that url path, afaik, returns the first 10 pages.